### DISCUSSION TITLE
Unable to lanuch the server .it is throwing session not created exception

### DISCUSSION THREAD

**Discussion by Lakshmanan4233** [appium/appium#18849] [Q&A]
In my circle ,my friend trying to connect with emulator (or) real device . But it is thrwing the errormessage.

I am just attached the error screenshot .

Here is my friend code snippet

public  class Base {

	public AndroidDriver driver;
	public AppiumDriverLocalService service;
	
	@Test
	public  void configureAppium() throws MalformedURLException
	{
		// Run appium server automatically

		service = new AppiumServiceBuilder()
				.withAppiumJS(new File("C:\\Users\\ciadmin\\AppData\\Roaming\\npm\\node_modules\\appium\\build\\lib\\main.js"))
				.withIPAddress("http://0.0.0.0")
				.usingPort(4723)
				.withTimeout(Duration.ofSeconds(60)).build();
		
		service.start();

		// Create Capabilities
		UiAutomator2Options options = new UiAutomator2Options();
		options.setDeviceName("TestDevice"); // Virtual Device Name
		options.setPlatformName("Android");
		options.setCapability("PlatformVersion", "12.0");
		options.setApp(System.getProperty("user.dir") + "\\src\\test\\java\\resources\\test.apk");
										
		// Create Object for Android driver / iOS driver.

		try {
			driver = new AndroidDriver(new URL("http:/0.0.0.0.4723"), options);
		} catch (MalformedURLException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
		
		//http://0.0.0.0:4723
		
		driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(60));
		driver.findElement(AppiumBy.accessibilityId("preference")).click();
		
		driver.quit();
		service.stop();

	}
}

My self also analyzed .But i can not able find it . Kindly hep me guys 

![Console](https://github.com/appium/appium/assets/96466534/cfa9743c-5cfc-4a17-802d-9e799b9591f4)

**Comment by jlipps** [comment]
It seems like this might be a question for the Java client repo

**Reply by Lakshmanan4233** [reply]
I am not understand your comments. Can you please explain more?

**Reply by KazuCocoa** [reply]
https://discuss.appium.io/ may help to ask questions, or https://github.com/appium/java-client if something issue existed.

Btw, `driver = new AndroidDriver(new URL("http:/0.0.0.0.4723"), options);` should be `driver = new AndroidDriver(new URL("http://0.0.0.0.4723"), options);`? (Missed one `/`)
